### DISCUSSION TITLE
Is Next.js patching the built-in Promise?

### DISCUSSION THREAD

**Discussion by brunouber** [vercel/next.js#52561] [App Router] [answered]
**Is Next.js patching the built-in Promise?**

We're using `openapi-typescript-codegen` to build our own api sdk from the backend openapi.json.
`openapi-typescript-codegen` produces a number of methods, eg. `client.postCreateUser`, that we then use along with react-query's `useMutation` hook.

`client.postCreateUser` returns a `CancelablePromise` - it extends built-in Promise.

Everything worked fine up until `next@3.4.7`.

After upgrading to `next@3.4.8 ` (tested aslo with `3.4.9` and `3.4.10.canary.3`) the Promise returned by `client.postCreateUser` never settles (it remains in pending state forever).
Nothing else changed in the repository - only Next.js upgrade.

That would only makes sense if Next.js is patching global Promise - is that the case?

**Comment by icyJoseph** [comment]
Yeah, something is not adding up:

- https://github.com/ferdikoomen/openapi-typescript-codegen/issues/1626
- https://github.com/ferdikoomen/openapi-typescript-codegen/pull/1627

I am gonna dig around, but I wonder if its an SWC related problem. Perhaps the code is mangled/minified, in an incorrect way 🤔 

@brunouber could you try with v13.4.8-canary.9 and v13.4.8-canary.10 - I want to discard an SWC issue.

**Reply by brunouber** [reply]
`13.4.8-canary.9` works fine.
`13.4.8-canary.10` introduces the issue.

**Reply by icyJoseph** [reply]
Awesome! There was an SWC upgrade in between those two!

**Reply by icyJoseph** [reply]
Now, how do we create a minimal reproduction so that the SWC folks can take a look 🤔 x 1000

**Reply by brunouber** [reply]
Please, could you tell exactly the change? 
I mean the SWC version that was used in canary.9 and the one adopted in canary.10.

**Reply by icyJoseph** [reply]
Well, we don't know for sure yet, but there is a good chance it is that:

- https://github.com/vercel/next.js/commit/484bdebc2468e14a677f71f18457722ba33a69c3
- https://github.com/vercel/next.js/compare/v13.4.7...v13.4.8 (expand until you find v13.4.8-canary.10)

Exactly what has changed I don't know yet either, since that's a Rust crate used to compile the code in our apps.

**Reply by brunouber** [reply]
Thank you for putting me in the right direction 👍

**Comment by brunouber** [comment, marked as answer]
For future readers: https://github.com/swc-project/swc/issues/7658
